packages/curl.rb

```diff
@@ -3,40 +3,69 @@
-  })
-  binary_sha256({
-    aarch64: '439fb6468782be5842b4f7a3c2854e5026d3eaf3a99023dd3d82b540458ec110',
-     armv7l: '439fb6468782be5842b4f7a3c2854e5026d3eaf3a99023dd3d82b540458ec110',
-       i686: '98479b4d6c3d9deee57fb261b58c2a327ebdfaa15a14eead2d979a26e2f2c74d',
-     x86_64: '07d49c2d5ad27ccaeba9f50b73b5c0078a1d0c57902797e0b5cc2e98d4c1f2b9'
-  })
+  source_url "https://github.com/curl/curl/releases/download/curl-#{@_ver.gsub('.','_')}/curl-#{@_ver}.tar.xz"
+  source_sha256 'be42766d5664a739c3974ee3dfbbcbe978a4ccb1fe628bb1d9b59ac79e445fb5'
 
   depends_on 'ca_certificates' => :build
   depends_on 'hashpipe' => :build
   depends_on 'libunbound' => :build
   depends_on 'musl' => :build
-  depends_on 'py3_pip' => :build
   depends_on 'rust' => :build
   depends_on 'valgrind' => :build
 
+  def self.prebuild
+    FileUtils.mkdir_p 'deproot/include'
+    FileUtils.mkdir_p 'deproot/lib'
+    @builddirs = ['zlib', 'zstd', 'brotli', 'idn2', 'ssl', 'libssh', 'nghttp2']
+    @builddirs.each do |builddir|
+      FileUtils.mkdir_p "build/#{builddir}"
+    end
+
+    puts "Downloading supporting libraries.".yellow
+    Dir.chdir 'build/zlib' do
+      system 'curl -#L https://zlib.net/zlib-1.2.11.tar.gz | \
+        hashpipe sha256 c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1 | \
+        tar xz --strip-components=1'
+    end
+    Dir.chdir 'build/zstd' do
+      system 'curl -#L https://github.com/facebook/zstd/archive/v1.5.0.tar.gz | \
+        hashpipe sha256 0d9ade222c64e912d6957b11c923e214e2e010a18f39bec102f572e693ba2867 | \
+        tar xz --strip-components=1'
+    end
+    Dir.chdir 'build/brotli' do
+      system 'curl -#L https://github.com/google/brotli/archive/v1.0.9.tar.gz | \
+        hashpipe sha256 f9e8d81d0405ba66d181529af42a3354f838c939095ff99930da6aa9cdf6fe46 | \
+        tar xz --strip-components=1'
+    end
+    Dir.chdir 'build/idn2' do
+      system 'curl -#L https://ftpmirror.gnu.org/libidn/libidn2-2.3.2.tar.gz | \
+        hashpipe sha256 76940cd4e778e8093579a9d195b25fff5e936e9dc6242068528b437a76764f91 | \
+        tar xz --strip-components=1'
+    end
+    Dir.chdir 'build/ssl' do
+      system 'curl -#L https://www.openssl.org/source/openssl-1.1.1k.tar.gz | \
+        hashpipe sha256 892a0875b9872acd04a9fde79b1f943075d5ea162415de3047c327df33fbaee5 | \
+        tar xz --strip-components=1'
+    end
+    Dir.chdir 'build/libssh' do
+      system 'curl -#L https://www.libssh.org/files/0.9/libssh-0.9.5.tar.xz | \
+       hashpipe sha256 acffef2da98e761fc1fd9c4fddde0f3af60ab44c4f5af05cd1b2d60a3fa08718 | \
+        tar xJ --strip-components=1'
+    end
+    Dir.chdir 'build/nghttp2' do
+      system 'curl -#L https://github.com/nghttp2/nghttp2/releases/download/v1.44.0/nghttp2-1.44.0.tar.bz2 | \
+        hashpipe sha256 989971276517a1c9ed330b779c34cf02d99da3b85d156eb297f42b1b7227b297 | \
+        tar xj --strip-components=1'
+    end
```

Comment:
Why not just use crew packages instead?

Reply:
Because these are for building against musl...

Reply:
Honestly I've been thinking of switching entirely to musl with Chromebrew. Like wouldn't that fix so many glibc problems and allow chromebrew, and allow binaries to be much smaller and run faster? @satmandu let me know you're opinion, but I'd really like to start doing that.

Reply:
This PR is way too complex.  It also won't build on i686.  Would it make sense to split it out into dependency packages?

Reply:
Sure... That makes sense.

Let me create some musl_ packages, which will put stuff in the non-FHS compliant (but standard for musl!) #{CREW_PREFIX}/musl 

What do you think @saltedcoffii ?

Reply:
I think we have much bigger priorities. Many packages don't compile with musl without void or alpine patches, so easy patch downloading is a must, hence the extra url PR. I also think we need (and have needed for a while) a way to track and remove orphaned dependencies, much like `apt autoremove`. Is that reasonable @satmandu ?